src/common/math/Vector.h

```diff
@@ -87,9 +90,9 @@ namespace Math {
 		void Store(Type* ptr) const;
 
 		// Apply a function on all elements of the vector
-		template<typename Func> Vector<Dimension, typename std::result_of<Func(Type)>::type> Apply(Func func) const;
-		template<typename Func, typename Type2> Vector<Dimension, typename std::result_of<Func(Type, Type2)>::type> Apply2(Func func, Vector<Dimension, Type2> other) const;
-		template<typename Func, typename Type2, typename Type3> Vector<Dimension, typename std::result_of<Func(Type, Type2, Type3)>::type> Apply3(Func func, Vector<Dimension, Type2> other1, Vector<Dimension, Type3> other2) const;
+		template<typename Func> Vector<Dimension, typename resultof_t<Func, Type>> Apply(Func func) const;
```

Comment:
Huh, apparently you have to remove the `typename` here. I always believed that spamming more `typename` never hurts!